nengo/tests/test_copy.py

```diff
@@ -0,0 +1,228 @@
+from copy import copy
+import warnings
+
+import numpy as np
+import pytest
+
+import nengo
+from nengo.exceptions import NetworkContextError, NotAddedToNetworkWarning
+from nengo.params import params
+from nengo.utils.compat import is_array_like, pickle
+from nengo.utils.testing import warns
+
+
+@pytest.fixture(scope='module', autouse=True)
+def error_on_not_added_to_network_warning(request):
+    catcher = warnings.catch_warnings(record=True)
+    catcher.__enter__()
+    warnings.simplefilter(
+        'error', category=NotAddedToNetworkWarning)
+    request.addfinalizer(lambda: catcher.__exit__(None, None, None))
+
+
+def assert_is_copy(cp, original):
```

Comment:
It's just a utility function because it doesn't start with `test_`

On May 24, 2016 6:36:33 PM GMT+01:00, Sean Aubin <notifications@github.com> wrote:
>> +from nengo.exceptions import NetworkContextError,
>NotAddedToNetworkWarning
>> +from nengo.params import params
>> +from nengo.utils.compat import is_array_like, pickle
>> +from nengo.utils.testing import warns
>> +
>> +
>> +@pytest.fixture(scope='module', autouse=True)
>> +def error_on_not_added_to_network_warning(request):
>> +    catcher = warnings.catch_warnings(record=True)
>> +    catcher.__enter__()
>> +    warnings.simplefilter(
>> +        'error', category=NotAddedToNetworkWarning)
>> +    request.addfinalizer(lambda: catcher.__exit__(None, None, None))
>> +
>> +
>> +def assert_is_copy(cp, original):
>
>This might just be me being a noob at `pytest`, but how are these
>parameters defined? I thought they had to be `pytest` fixtures? I see
>that `original` is paramterized at the bottom of the file. Or are these
>just utility functions? What's the distinction between a utility
>function and a test?
>
>---
>You are receiving this because you were assigned.
>Reply to this email directly or view it on GitHub:
>https://github.com/nengo/nengo/pull/984/files/465d6a9e7b01caece896adee1916793911f9e5d2#r64436948

-- 
Sent from my Android device with K-9 Mail. Please excuse my brevity.